### PR TITLE
Fix a bug in converting LIKE patterns to regexes.

### PR DESCRIPTION
**Pull request by sellout**
This is probably not a full fix to SD-1081, but this bug would be hit
once the other bug gets fixed.

### CODE REVIEW

File: core/src/main/scala/quasar/compiler.scala [hunk 1]
```diff
@@ -193,10 +193,10 @@ trait Compiler[F[_]] {
         String = {
       def sansEscape(pat: List[Char]): List[Char] = pat match {
         case '_' :: t =>         '.' +: escape(t)
-        case '%' :: t => ".*".toList ++ escape(t)
-        case c :: t   =>
+        case '%' :: t => ".*".toList ⊹ escape(t)
```
Comment (mossprescott) [inline core/src/main/scala/quasar/compiler.scala]: First time I've seen this, although I see we have a definition for `SemigroupF` (which we'e apparently not using). Any particular reason to switch? Just curious.

**Comment by mossprescott** [comment]
Sorry, just noticed this PR hanging out there. Just the one question, otherwise obviously this is a good fix for an embarrassing bug.

File: core/src/main/scala/quasar/compiler.scala (hunk 1 above)
Comment (sellout) [inline core/src/main/scala/quasar/compiler.scala]: I just tend to prefer more general operations. In this case, I think the function could avoid the conversions to/from List and work directly on String, but I didn’t take that step. So it’s just a little flag like “oh, this just needs a semigroup, and we have one of those for strings …”

We do use it in a few other places, I think, but maybe not the most compelling reason.

**Comment by jdegoes** [comment]
This would be good to get in 2.2.2

**Pull request merged**